### ISSUE TITLE
Inconsistent behavior with Redis in JSON.MSET command

### ISSUE THREAD

**Issue by VIVALXH** [apache/kvrocks#2552]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Version

redis:
image: redislabs/rejson:latest

kvrocks:
image: apache/kvrocks:latest

### Minimal reproduce step

redis:
image: redislabs/rejson:latest
``` 
❯ redis-cli -h 192.168.1.72 -p 6369
192.168.1.72:6369> flushdb
OK
192.168.1.72:6369> JSON.SET doc $ '{"f1":{"a1":0},"f2":{"a2":0}}'
OK
192.168.1.72:6369> JSON.GET doc
"{\"f1\":{\"a1\":0},\"f2\":{\"a2\":0}}"
192.168.1.72:6369> JSON.MSET doc $.f1.a1 1 doc $.f2.a2 2
OK
192.168.1.72:6369> JSON.GET doc
"{\"f1\":{\"a1\":1},\"f2\":{\"a2\":2}}"
```

kvrocks:
image: apache/kvrocks:latest
``` 
❯ redis-cli -h 192.168.1.72 -p 6666
192.168.1.72:6666> flushdb
OK
192.168.1.72:6666> JSON.SET doc $ '{"f1":{"a1":0},"f2":{"a2":0}}'
OK
192.168.1.72:6666> JSON.GET doc
"{\"f1\":{\"a1\":0},\"f2\":{\"a2\":0}}"
192.168.1.72:6666> JSON.MSET doc $.f1.a1 1 doc $.f2.a2 2
OK
192.168.1.72:6666> JSON.GET doc
"{\"f1\":{\"a1\":0},\"f2\":{\"a2\":2}}"
``` 

### What did you expect to see?

 I hope that the behavior of JSON.MSET in KVROCKS is consistent with Redis.

### What did you see instead?

For Redis, the JSON.MSET command can modify multiple fields of the same key simultaneously, but for KVROCKS, only one field is successfully modified.

### Anything Else?
I found that in the code, JSON.MSET performs a query for each sub-command, resulting in a separate result in memory for each key if the keys are the same. Can it be modified so that there is only one copy of the key in memory and no repeated reads?
https://github.com/apache/kvrocks/blob/f2bc224bc342f7e2b679cb2dc79d7a31067643ca/src/types/redis_json.cc#L579C15-L579C17

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

**Comment by git-hulk** [comment]
@VIVALXH Good catch, would you like to fix this?

**Comment by VIVALXH** [comment]
Thank you for your response and for recognizing the issue. I'd be happy to try and fix this issue, and I plan to start working after my workday.

**Comment by git-hulk** [comment]
@VIVALXH Thank you!

**Comment by PragmaTwice** [comment]
Fixed in #2579.
